src/wallet/wallet-generation.ts

```diff
@@ -0,0 +1,198 @@
+import axios from 'axios'
+
+import {RippleAPI} from '..'
+import {errors} from '../common'
+import {GeneratedAddress} from '../offline/generate-address'
+import {isValidAddress} from '../common/schema-validator'
+import {reject} from 'lodash'
+
+export interface FaucetWallet {
+  account: GeneratedAddress
+  amount: number
+  balance: number
+}
+
+export enum FaucetNetwork {
+  Testnet = 'https://faucet.altnet.rippletest.net/accounts',
+  Devnet = 'https://faucet.devnet.rippletest.net/accounts'
+}
+
+const INTERVAL_SECONDS = 1 // Interval to check an account balance
+const MAX_ATTEMPS = 20 // Maximum attemptsto retrieve a balance
+
+/**
+ * Generates a random wallet with some amount of XRP (usually 1000 XRP).
+ *
+ * @param wallet - An existing XRPL wallet to fund, if undefined, a new wallet will be created.
+ * @returns - A Wallet on the Testnet or Devnet that contains some amount of XRP.
+ */
+async function generateFaucetWallet(
+  this: RippleAPI,
+  wallet?: string
```

Comment:
If `wallet` is just supposed to be the address, I think it should be renamed to something like `address` - `wallet` implies it has something to do with the keys

Reply:
Do we want to add an optional `url?: string` field that allows users to pass in their own Faucet URL? For use w/ other faucets (like a Hooks devnet faucet).

Reply:
I'll rename `wallet` to `address`.

I don't mind adding a `url` field as well.